### PR TITLE
feat(kgo): use emptyDir as /tmp

### PR DESCRIPTION
**Pull request by tao12345666333**
#### What this PR does / why we need it:

use emptyDir to satisfy the read-only root FS requirement.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes https://github.com/Kong/gateway-operator/issues/1323

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] PR is based off the current tip of the `main` branch.
- [ ] Changes are documented under the "Unreleased" header in CHANGELOG.md
- [ ] New or modified sections of values.yaml are documented in the README.md
- [x] Commits follow the [Kong commit message guidelines](https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#commit-message-format)

### CODE REVIEW

File: charts/gateway-operator/templates/_helpers.tpl [hunk 1]
```diff
@@ -59,3 +59,14 @@ app.kubernetes.io/instance: "{{ .Release.Name }}"
...
+
+{{- define "kong.volumes" -}}
+- name: {{ template "kong.fullname" . }}-tmp
+  emptyDir:
+    sizeLimit: {{ .Values.tmpDir.sizeLimit }}
+{{- end }}
+
+{{- define "kong.volumeMounts" -}}
+- name: {{ template "kong.fullname" . }}-tmp
+  mountPath: /tmp/k8s-webhook-server/serving-certs
```
Comment (randmonkey) [inline charts/gateway-operator/templates/_helpers.tpl:71]: I think the combination of volume name and the path/usage of the volume may be confusing:
- `{{ kong.fullname }}-tmp` means all tmp files are added into this dir
- the mountPath means that the dir is only for webhook certs
Do you have a better naming strategy?

File: charts/gateway-operator/values.yaml [hunk 2]
```diff
@@ -64,3 +64,7 @@ controllers:
 
 # Use this section to add environment variables to operator's container
 env: {}
+
+# Use this section to change the tmp empty dir size
+tmpDir:
+  sizeLimit: 256Mi
```
Comment (randmonkey) [inline charts/gateway-operator/values.yaml:70]: The `256Mi` is just an arbitrary value, or any standards/best practices for this value?

File: charts/gateway-operator/templates/_helpers.tpl (hunk 1 above)
Comment (tao12345666333) [reply charts/gateway-operator/templates/_helpers.tpl:71]: `certs-dir`?

Comment (randmonkey) [reply charts/gateway-operator/templates/_helpers.tpl:71]: sounds good.

File: charts/gateway-operator/values.yaml (hunk 2 above)
Comment (tao12345666333) [reply charts/gateway-operator/values.yaml:70]: This directory is used for storing KGO's admission certificates.
Typically, only a tiny space is needed.  https://github.com/Kong/gateway-operator/blob/1e36e7d9381727c65a399dcbf2faf2bcd6dff79e/modules/manager/webhook.go#L130-L148


There is no definitive best practice here; I simply chose a value that would have a relatively minimal negative impact.

File: charts/gateway-operator/templates/_helpers.tpl [hunk 3]
```diff
@@ -59,3 +59,14 @@ app.kubernetes.io/instance: "{{ .Release.Name }}"
   value: {{ $val | quote }}
 {{- end -}}
 {{- end -}}
+
+{{- define "kong.volumes" -}}
+- name: {{ template "kong.fullname" . }}-certs-dir
+  emptyDir:
+    sizeLimit: {{ .Values.tmpDir.sizeLimit }}
```
Comment (randmonkey) [inline charts/gateway-operator/templates/_helpers.tpl]: the key here is not changed while renaming.
```suggestion
    sizeLimit: {{ .Values.certsDir.sizeLimit }}
```

**Review by randmonkey — APPROVED** [review]

**Pull request merged**